### DISCUSSION TITLE
What would be needed for ethernet?

### DISCUSSION THREAD

**Discussion by rudybernard** [trombik/esp_wireguard#40] [Ideas]
Hi, I have tested wireguard on esp32 wifi and it is working amazingly well!! I know that in the README.md, it is a known issue that ethernet is not supported, but I would like to know if anybody have  idea where to start to make it work on ethernet. I will investiguate a solution on my side and make a pull request if I make it work, but any help will be appreciated. 

Thanks!

**Comment by rudybernard** [comment]
I'm looking at the wirefuardif.c file and line 924 seem to be an important one. I tought it could fix it all with one small change since it is kind of a selection the TCPIP adapter, but it doesn't change anything.

original
`tcpip_adapter_get_netif(TCPIP_ADAPTER_IF_STA, &underlying_netif);`

could be change to this but no effect
`tcpip_adapter_get_netif(TCPIP_ADAPTER_IF_ETH, &underlying_netif);`

**Reply by Odysseusfr** [reply]
Which esp-idf version are you using ?

**Comment by rudybernard** [comment]
I use platformio and the framework i'm using is 
framework-arduinoespressif32 @ 3.20003.220626 (2.0.3)
